Makefile

```diff
@@ -1,14 +1,15 @@
 SDCC ?= sdcc
-STCCODESIZE ?= 4089
-SDCCOPTS ?= --code-size $(STCCODESIZE) --xram-size 0 --data-loc 0x30 --disable-warning 126 --disable-warning 59
-SDCCREV ?= -Dstc15f204ea
+STCCODESIZE ?= 8184
```

Comment:
currently we have about three different mcu variants:
1. stc15f204ea (4k code, 256b ram, no xram)
2. stc15w404as (4k code, 256b ram, 256b xram)
3. stc15w408as (8k code, 256b ram, 256b xram).
So while the stack/ram size opts are generally ok for 4xx series, they wont work properly on 2xx series.

Additionally complicating this, is that while there is this define in the code, e.g. "stc15w408as", it was unfortunately used to mean both the mcu and some options of certain clock hardware options (e.g. 3rd button). What I really need to do is to separate out the mcu options from the hardware variant options (in hwconfig.h) as the mcus are generally interchangeable, while the hardware options are not.